### PR TITLE
🐛 Fix Process ID Check in ID Validation

### LINKED ISSUE #1948: Modeler Deletes all Elements
## Description

Jan-22-2020 09-24-12

The Modeler doesn't show any elements in the lane anymore.

## Reproduction

1. Open two processes
2. Set the Id of the "Collaboration" to the name of the "Participant"
3. The PropertyPanel signals a duplicate Id by adding a red border now
4. Now, without doing anything else, navigate to the second process
5. Navigate back to the first process

## Setup

BPMN-Studio: 5.11.2

### PR DESCRIPTION
**Pull request by SteffenKn**
## Changes

1. Fix Process ID Check in ID Validation

## Issues

Closes #1948 

PR: #1949

## How to test the changes

- Create a new diagram
- Copy the pool id
- Use the pool id as id for any other element
- Change the diagram (without clicking anywhere else first)
- Change back to the previous diagram
- **Notice that the diagram is still completely visible.**
- Copy the collaboration id
- Use the collaboration id as process id
- Change the diagram (without clicking anywhere else first)
- Change back to the previous diagram
- **Notice that the diagram is still completely visible.**

### CODE REVIEW

File: src/modules/design/property-panel/indextabs/general/sections/basics/basics.ts [hunk 1]
```diff
@@ -239,14 +239,16 @@ export class BasicsSection implements ISection {
...
-  private isProcessIdUnique(id: string): boolean {
+  private areRootElementIdsUnique(id: string): boolean {
     // eslint-disable-next-line no-underscore-dangle
     const elementIds: Array<string> = this.modeler._definitions.rootElements.map((rootElement: IModdleElement) => {
       return rootElement.id;
     });
 
-    const currentId: number = elementIds.indexOf(id);
-    elementIds.splice(currentId, 1);
+    const currentId: number = elementIds.indexOf(this.businessObjInPanel.id);
```
Comment (ElRaptorus) [inline src/modules/design/property-panel/indextabs/general/sections/basics/basics.ts]: Man sollte die Variable als "Index" kennzeichnen, damit nicht der Eindruck entsteht, man hätte hier eine tatsächliche ID in der Hand.

```suggestion
    const currentElementIdIndex = elementIds.indexOf(this.businessObjInPanel.id);
```

File: src/modules/design/property-panel/indextabs/general/sections/basics/basics.ts [hunk 2]
```diff
@@ -239,14 +239,16 @@ export class BasicsSection implements ISection {
...
+    if (currentId >= 0) {
```
Comment (ElRaptorus) [inline src/modules/design/property-panel/indextabs/general/sections/basics/basics.ts]:
```suggestion
    if (currentElementIdIndex >= 0) {
```

File: src/modules/design/property-panel/indextabs/general/sections/basics/basics.ts [hunk 3]
```diff
@@ -239,14 +239,16 @@ export class BasicsSection implements ISection {
...
+      elementIds.splice(currentId, 1);
```
Comment (ElRaptorus) [inline src/modules/design/property-panel/indextabs/general/sections/basics/basics.ts]:
```suggestion
      elementIds.splice(currentElementIdIndex, 1);
```

File: src/modules/design/property-panel/indextabs/general/sections/basics/basics.ts (hunk 1 above)
Comment (SteffenKn) [reply src/modules/design/property-panel/indextabs/general/sections/basics/basics.ts]: Stimmmt habe ich übersehen 👍

**Pull request merged**